### ISSUE TITLE
Uebersicht no longer takes full screen height on latest Sonoma beta

### ISSUE THREAD

**Issue by andrefgneves** [felixhageloh/uebersicht#517]
So, I think Apple changed something in the latest Sonoma beta, 14.1 Beta (23B5067a), that makes Uebersicht not have the full screen height on notched macs. The OS might be reporting the "safe" screen height (that subtracts the notch height from full screen height).

I tried setting a negative top on my widgets, but they get clipped.
Is there anything we can do to work around this? Or an update is required to fix it?

Edit: forgot to mention that I have my menu bar set to auto hide. Not sure that's relevant or not.

Here are some screenshots, if it helps clarify:

<img width="1799" alt="1" src="https://github.com/felixhageloh/uebersicht/assets/3869532/7b010b69-1823-4c92-af34-01a34c3db571">

<img width="807" alt="2" src="https://github.com/felixhageloh/uebersicht/assets/3869532/b0f6bf23-1efa-4557-95c8-c4ee7642ad60">

**Comment by Larcaster** [comment]
i have the same problem

**Comment by BorisAnthony** [comment]
Still true after Sonoma release.
Restarting Übersicht fixes it, but it returns when switching primary monitors (i.e.: I unplug the macbook from the monitor, which triggers of course screen size change).
<img width="1802" alt="Screenshot 2023-10-24 at 06 44 44" src="https://github.com/felixhageloh/uebersicht/assets/964649/2d5b2a0e-7d08-49ec-ba2f-20ecfb566c1e">

**Comment by max1mvs** [comment]
> Still true after Sonoma release. And it goes further…

I can confirm this. All of my positioned widgets are now offset by the height of the regular menu bar (I have it only to show when I mouse over it permanently).

https://github.com/felixhageloh/uebersicht/assets/17984541/d6723c51-41c9-4aa4-b3a1-cf90f28af812

**Comment by felixhageloh** [comment]
I can reproduce this as well! I am almost done fixing #516 and then I will look into this one

**Comment by matzegebbe** [comment]
Thanks @felixhageloh Because of this bugfix I became a fan

**Comment by jrucho** [comment]
Same problem here.
![Screenshot 2023-11-01 at 12 39 45](https://github.com/felixhageloh/uebersicht/assets/22379429/116c8815-5305-4435-adb1-36d3f289d0b5)

**Comment by Helloyunho** [comment]
![CleanShot 2023-11-08 at 22 49 22@2x](https://github.com/felixhageloh/uebersicht/assets/10192625/d93ebf77-1f8a-4591-b1d0-33ed0b9367f6)
so I updated Ubersicht and I'm having the exact same problem... it's been a month since this issue was created, any solutions? I think I'm gonna use the older one for now

update: the older version doesn't work properly either... I just updated to 14.1 so maybe that's the problem..?

**Comment by GiyoMoon** [comment]
Yes, updating to macOS 14+ caused this issue to appear for me too :/

**Comment by felixhageloh** [comment]
Alright, fixing #516 took a bit longer than expected. 
I have a new version here http://tracesof.net/uebersicht/releases/Uebersicht-1.6.78.app.zip that should fix this issue. Please let me know if does for you as well!

**Comment by andrefgneves** [comment]
> that should fix this issue

It does. Thank you!

**Comment by max1mvs** [comment]
Thank you for the update! It works fine on my external monitor but not on my MacBook…

<img width="573" alt="Screenshot 2023-11-09 at 10 45 43" src="https://github.com/felixhageloh/uebersicht/assets/17984541/50def6a1-2f15-472e-b72b-67254e3b09d0">

It "cuts off" right under the notch:

![IMG_1866](https://github.com/felixhageloh/uebersicht/assets/17984541/dc95ed30-43c5-4168-aa4a-56d866ad5268)

**Comment by JakubMarkiewicz** [comment]
> Thank you for the update! It works fine on my external monitor but not on my MacBook…
> 

Can confirm. Works great on external monitor, same behaviour on macbook with notch

**Comment by felixhageloh** [comment]
Interesting - I have notch as well and I am not seeing the same issue. Maybe I am understanding it wrong though. Can you elaborate a bit?

**Comment by GiyoMoon** [comment]
Same for me. I have a MacBook Pro 14-inch 2021 with Sonoma 14.1
Screenshot of macbook screen:
<img width="1511" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/36984337/8a766522-2ed7-49c7-81a1-1195446d7bcf">
External monitor:
<img width="2560" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/36984337/efe3f207-b868-4a65-9705-b3956f343816">

**Comment by jrucho** [comment]
> Alright, fixing #516 took a bit longer than expected. I have a new version here http://tracesof.net/uebersicht/releases/Uebersicht-1.6.78.app.zip that should fix this issue. Please let me know if does for you as well!

It fixed the issue on my side 👍

**Comment by HeySora** [comment]
Can confirm it did not fix the issue when using the MacBook without an external screen :(

Here is how it looks like with the pinned macOS menu bar enabled and disabled

Pinned menu bar enabled: (working perfectly fine)
![image](https://github.com/felixhageloh/uebersicht/assets/17962248/c27dbb39-d64e-45a9-a357-895f60a974f1)

Pinned menu bar disabled: (Notch gap still present above. Übersicht's height still increases a bit but still doesn't fully extend)
![image](https://github.com/felixhageloh/uebersicht/assets/17962248/4b162aef-f2b0-4cbc-97eb-086c534f090d)

IRL picture showing the issue:
![IMG_2196](https://github.com/felixhageloh/uebersicht/assets/17962248/a9c55ebd-a02b-49a9-b6c0-6025d1d2d76f)

Clearing the macOS Saved Application State for Übersicht (which stores the last known window positions) as well as the caches did not fix anything either.

**Comment by Helloyunho** [comment]
I tried the updated one but the issue is still there... though on the external display, it looks perfect again! only my MacBook screen which has a notch on top has the problem..

**Comment by felixhageloh** [comment]
Thank you for the detailed report @HeySora, I get the problem now. Also thanks everyone else! I will have a look again

**Comment by max1mvs** [comment]
> Thank you for the detailed report @HeySora, I get the problem now. Also thanks everyone else! I will have a look again

Yeah, I can see that my screenshots are a tad confusing – it looks like the menu bar is enabled, which it is not. I always have it off, and I have a dimmed layer on top of my background using Übersicht - my bad! 😅

**Comment by felixhageloh** [comment]
Ok I have a new build - not completely done testing and cleaning up the code, but I want to check if this is going in the right direction https://tracesof.net/uebersicht/releases/Uebersicht-1.6.79.app.zip

**Comment by artmoskal** [comment]
Last one (Uebersicht-1.6.79.app.zip) fine on external mon but has gap on macbook (notch). Also unplugging external monitor causing error:
<img width="658" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/107350106/523177a8-7572-4874-9900-31fe21695eb0">
And I can't start uebersicht any more unless second mon is plugged back (trying to restart app gives the same error). 

Also it it is bigger deal that it seems (at least for me) because Yabai thinks offset is same on both monitors so to make top part of the window visible on my macbook I need to add gap on external mon. Please check screnshot:
<img width="630" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/107350106/b694b0aa-7678-4365-af91-c0bc99434dc3">

P.S. Last unplug did not cause error but 
 1. second monitor's widgets appear in the middle of the screen (so two panels on the screen)
 2. first monitor's widget bar stuck to very top (e.g. "top gap issue" "disappeared") 
 3. looks like widget started to cover system panel as I was not able to make the standard panel visible by moving mouse to the top of the screen

**Comment by HeySora** [comment]
@felixhageloh Hm, it did work flawlessly for 10 seconds!! but as soon as I switched spaces, it moved down :sob:

I __guess__ it is progress in the right direction! I cannot make it work anymore though. Sometimes refreshing all widgets makes them positioned correctly again, but this is highly inconsistent.

--

Actually, nevermind! `rm -r ~/Library/Caches/tracesOf.Uebersicht` fixed the issue on my end! It was probably an odd quirk coming from running the previous 77 and 78 builds.

I cannot __extensively__ test multiple monitors right now, but so far it seems stable! Thank you for the amazing work 🙏 I'll report in a day or two to report the experience.

**Comment by felixhageloh** [comment]
thank you @HeySora! Interesting that you had to clear caches 🤔. @artmoskal does clearing the cache solve the issue for you as well?

**Comment by ghost** [comment]
It didn't work for me. Mac with notch. No change after clearing the cache

**Comment by tubenhirn** [comment]
@Larcaster for me it also didn't work in the beginning. Clearing the cache folder had no effect at all. I switched the menu-bar configuration (hide / show) multiple times and voilà, the widget snaped to the top.

**Comment by artmoskal** [comment]
> thank you @HeySora! Interesting that you had to clear caches 🤔. @artmoskal does clearing the cache solve the issue for you as well?

 clearing the cache does address some issue(s), such as the one on the white background related to "assignment blabla", and another issue involving two widgets appearing on the screen (which had same steps to reproduce). However, the "notch" issue persists. To summarize - widgets now function appropriately, except for the main monitor widget, which is still erroneously shifted to the bottom.

If further details, logs, or experimental results might be of assistance, please don't hesitate to inform me.

P.S. Reflecting on it, I recall encountering a similar issue with either two widgets or a widget "escaping" from the top of the screen. Although this seemed rare and was typically resolved with a simple restart of the application, I didn't pay much attention to it at the time.

**Comment by max1mvs** [comment]
> Ok I have a new build - not completely done testing and cleaning up the code, but I want to check if this is going in the right direction https://tracesof.net/uebersicht/releases/Uebersicht-1.6.79.app.zip

Did, unfortunately, not work for me. And not with clearing the cache, either.

**Comment by felixhageloh** [comment]
@artmoskal  & @max1mvs  can you also try switching the menu bar auto hide behavior back and forth in the system settings (while Übersicht is running)?

<img width="478" alt="Screenshot 2023-11-13 at 09 42 08" src="https://github.com/felixhageloh/uebersicht/assets/367503/fe8e2fbd-800d-48d3-8a0d-c39b68bb7eb3">

**Comment by ghost** [comment]
After a few toggles, the widget took the right place

**Comment by artmoskal** [comment]
Tried:
1. It overlaps widgets (left part is macbook, right is monitor - widget panel completely overlapped)
<img width="1115" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/107350106/d3030b70-2501-443e-a5a6-b1e4c72fb1c9">

Then:
2. Unplugging external mon gives "assignment blabla" error BUT after switching spaces application appear on the intended place and after plugging monitor back it behaves correctly:
<img width="305" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/107350106/4fb7b5c3-cb72-47fd-a29c-6bb6dbe012b9">

Then:
3. Switching  back "Automatically hide and show menu bar" to "Always" fixes everything (including notch issue, no gap on MacBook mon):
<img width="483" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/107350106/164417f0-bdb1-4fce-a70a-e4d0e0237817">

4. Now even after app restart it works as intended. Will try to reboot and update this comment. 

---
Reboot returns "notch" issue. Playing around "Automatically hide" resolves it (temporarily). @felixhageloh

**Comment by max1mvs** [comment]
> @artmoskal & @max1mvs can you also try switching the menu bar auto hide behavior back and forth in the system settings (while Übersicht is running)?
> 
> <img alt="Screenshot 2023-11-13 at 09 42 08" width="478" src="https://user-images.githubusercontent.com/367503/282407023-fe8e2fbd-800d-48d3-8a0d-c39b68bb7eb3.png">

My goodness, that worked! 😁🕺 – I haven't tried rebooting… I will check that soon.

----- EDIT -----

Rebooting made the issue appear again… 😅

**Comment by pshomov** [comment]
I have Sonoma 14.1.1 on a 16' pro model and build 79 is looking pretty good. Have not tried external monitor yet.

**Comment by eagleyez** [comment]
> > @artmoskal & @max1mvs can you also try switching the menu bar auto hide behavior back and forth in the system settings (while Übersicht is running)?
> > <img alt="Screenshot 2023-11-13 at 09 42 08" width="478" src="https://user-images.githubusercontent.com/367503/282407023-fe8e2fbd-800d-48d3-8a0d-c39b68bb7eb3.png">
> 
> My goodness, that worked! 😁🕺 – I haven't tried rebooting… I will check that soon.
> 
> ----- EDIT -----
> 
> Rebooting made the issue appear again… 😅

it is not just rebooting, sometimes going full screen and returning on apps is also causing the issue to re-appear again. But yes, the workaround for turning showing and hiding menu bar (I have to do it atleast twice) works as a temporary workaround on 1.6 (79)

**Comment by felixhageloh** [comment]
Sorry for the late update, it was a hectic week. It also turns out to be more fiddly than I expected and I think Apple is still actively patching things. I noticed some API behave differently between 14.0 and 14.1. 

Anyway, I think I might have found the right approach now now: https://tracesof.net/uebersicht/releases/Uebersicht-1.6.80.app.zip

Please let me know how this works for you!

**Comment by matzegebbe** [comment]
LGTM! It worked out of the box with `1.6.80` on my 16" MBP M2 with a notch and two external monitors as well! Thanks, @felixhageloh!

**Comment by eagleyez** [comment]
Works for me too on 1.6.80... Thanks @felixhageloh for the quick turnaround!

**Comment by Helloyunho** [comment]
so far so good! thanks for your great work @felixhageloh!

**Comment by andrewheiss** [comment]
I don't know if this is related, but with 1.6.80 on macOS 14.1.1, the menu bar resizing issue has been fixed for me, but now all the widgets that I position using `bottom` are relative to the *top* of the dock. This didn't happen with 1.6.77, so it might be related?

For instance, I have a weather widget with this set in its coffeescript file:

```
style: """
  bottom: 0px
  ...
"""
```

When the dock is pinned open, the bottom of the widget is aligned with the top of the dock:

<img width="554" alt="image" src="https://github.com/felixhageloh/uebersicht/assets/73663/f158772f-b062-4c62-b509-84a48aff78fa">

When I hide the dock, the widget moves back to the bottom:

<img width="511" alt="Screenshot 2023-11-21 at 00 48 27" src="https://github.com/felixhageloh/uebersicht/assets/73663/0259bf50-c828-40bd-84f8-8843e0982a20">

Here's what it looks like in real time:

https://github.com/felixhageloh/uebersicht/assets/73663/ea0157be-ff98-41f3-b05f-23bbe3899645

If the dock is hidden and I move the mouse to show it, the widget stays pinned at the bottom of the screen permanently (as expected), even when the dock pops up:

https://github.com/felixhageloh/uebersicht/assets/73663/0ee15cbf-843f-41b0-a551-f92db638ebdc

**Comment by max1mvs** [comment]
> Sorry for the late update, it was a hectic week. It also turns out to be more fiddly than I expected and I think Apple is still actively patching things. I noticed some API behave differently between 14.0 and 14.1.
> 
> Anyway, I think I might have found the right approach now now: https://tracesof.net/uebersicht/releases/Uebersicht-1.6.80.app.zip
> 
> Please let me know how this works for you!

Seems to work for me too! Awesome job! Thanks! 😄⭐️👍🕺🎉

**Comment by GiyoMoon** [comment]
> Sorry for the late update, it was a hectic week. It also turns out to be more fiddly than I expected and I think Apple is still actively patching things. I noticed some API behave differently between 14.0 and 14.1.
> 
> Anyway, I think I might have found the right approach now now: https://tracesof.net/uebersicht/releases/Uebersicht-1.6.80.app.zip
> 
> Please let me know how this works for you!

Works for me too! Many many thanks for your work and quick fix 🙏 🎉

**Comment by iminlikewithyou** [comment]
still not working right for me on uebersicht 1.6.80 and macOS 14.1.1 - 
![CleanShot 2023-11-22 at 21 36 19@2x](https://github.com/felixhageloh/uebersicht/assets/20297536/9f1052fd-a04f-4a57-b892-52208803dd08)
(sorry for my black background)

seems like the top margin is still there but there is also an additional large bottom margin from the dock which wasnt there before.

edit: when I change the menu bar in System Preferences, i see the top margin moving up and down but it's only by a couple pixels. it seems like the notch might be getting included in the margin? i'm on an M2 Mac which does have a notch but this wasn't how it was working previously.

**Comment by felixhageloh** [comment]
thanks @iminlikewithyou , I forgot to take into account the Dock! This version should fix that as well  https://tracesof.net/uebersicht/releases/Uebersicht-1.6.81.app.zip. Let me know if it works for you

**Comment by andrewheiss** [comment]
Awesome, that fixes it—thanks!

**Comment by iminlikewithyou** [comment]
really awesome :) thanks

**Comment by Nasto** [comment]
Thank you for your awesome, quick work on this, @felixhageloh !
Do you know when this new version will be released via homebrew? I got some "non-tech" friends and colleagues asking me about that issue and I'd like to tell them to just update via brew to fix it.

**Comment by felixhageloh** [comment]
@Nasto I've just officially released the update!

**Comment by felixhageloh** [comment]
thanks everyone again for your help!

**Comment by max1mvs** [comment]
> thanks everyone again for your help!

Thank -> you! 😄

**Comment by GiyoMoon** [comment]
@felixhageloh Übersicht is still on version 1.6.77 on the website and on brew? 🤔

**Comment by felixhageloh** [comment]
@GiyoMoon I'll update the main website link this weekend - I usually do that a few days after the release. The upgrade is already available through the in-app upgrade mechanism tho. 

I don't maintain the homebrew formula
